WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift

```diff
@@ -489,6 +527,43 @@ private extension AztecPostViewController {
 }
 
 
+
+// MARK: - Publish Button Methods
+extension AztecPostViewController: PostEditorStateContextDelegate {
+    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
+        if keyPath == #keyPath(AbstractPost.status), let status = post.status {
+            postEditorStateContext.updated(postStatus: PostStatus(rawValue: status) ?? .draft)
+        } else if keyPath == #keyPath(AbstractPost.dateCreated) {
+            let dateCreated = post.dateCreated ?? Date()
+            postEditorStateContext.updated(publishDate: dateCreated)
+        } else if keyPath == #keyPath(AbstractPost.content) {
+            let characterCount = post.content?.characters.count ?? 0
+            postEditorStateContext.updated(hasContent: characterCount > 0)
+        }
+    }
+
+    internal func context(_ context: PostEditorStateContext, didChangeAction: PostEditorAction) {
+        publishButton.title = context.publishButtonText
```

Comment:
In both of these delegate methods, is there a reason you're going via the context instead of using the passed in action / bool?